addons/udes_stock/models/product_product.py

```diff
@@ -134,3 +135,61 @@ def write(self, vals):
         check_upper_case_validation(self._name, self.env.user, vals)
         return super().write(vals)
 
+    def get_barcode(self):
+        """
+        Get barcode of the product depending on configuration
+        """
+        self.ensure_one()
+        if self.u_multiple_barcodes:
+            barcode = self.u_barcode_ids and self.u_barcode_ids[0].name or None
+        else:
+            barcode = self.barcode
+        return barcode
+
+    def get_list_barcodes(self):
```

Comment:
What do you think about `get_all_barcodes` for the name of this method?  I think it would make it clearer that we are getting all the barcodes for a product, rather than just some list of barcodes?

Reply:
+1